lib/iris/common/metadata.py

```diff
@@ -140,6 +140,88 @@ def __new__(mcs, name, bases, namespace):
         return super().__new__(mcs, name, bases, namespace)
 
 
+#
+# Dictionary operations for dealing with the CubeAttrsDict "split"-style attribute
+# dictionaries.
+#
+# The idea here is to convert a split-dictionary into a "plain" one for calculations,
+# whose keys are all pairs of the form ('global', <keyname>) or ('local', <keyname>).
+# And to convert back again after the operation, if the result is a dictionary.
+#
+# For "strict" operations this clearly does all that is needed.  For lenient ones,
+# we _might_ want for local+global attributes of the same name to interact.
+# However, on careful consideration, it seems that this is not actually desirable for
+# any of the common-metadata operations.
+# So, we simply treat "global" and "local" attributes of the same name as entirely
+# independent. Which happily is also the easiest to code, and to explain.
+#
+def xd_is_split(dic):
+    """Detect whether a dictionary is a "split-attribute" type."""
+    return hasattr(dic, "globals") and hasattr(dic, "locals")
+
+
+def xd_to_normal(dic):
+    """
+    Convert the input to a 'normal' dict with paired keys, if it is split-attrs type
+    """
+
+    def _global_then_local_items(dic):
+        # Routine to produce global, then local 'items' in order, and with all keys
+        # "labelled" as local or global type, to ensure they are all unique.
+        for key, value in dic.globals.items():
+            yield ("global", key), value
+        for key, value in dic.locals.items():
+            yield ("local", key), value
```

Comment:
`"global"` vs `"local"` seems an intended use case for [enum](https://docs.python.org/3/library/enum.html), instead of working with straight strings. But I'll let you decide if that's overkill.